### ISSUE TITLE
Downgrading in Stripe should add credit to user's account

### ISSUE THREAD

**Issue by carter2422** [restrictcontentpro/restrict-content-pro#442]
If you downgrade from, for example, a yearly plan to a monthly plan, all goes as expected, with today's payment being free. However, subsequent renewals will be charged at the normal rate rather than pulling from the left over credit of the previous yearly subscription. 

It seems to me that a subscription change where the proration credit is greater than the current charge should add any left over to the user's `Balance` in Stripe. Doing this would ensure that renewal payments would continue to pull from the credit until it was emptied. 

*Tested in 2.5.*

**Comment by pippinsplugins** [comment]
We can't really support this because we can't do it across all gateways. Had a long discussion about it recently and determined the only way to make it the same across all gateways is to simply discount the registration, even if that means funds get "left over".

**Comment by mindctrl** [comment]
I'm wondering if we should change the way we think about downgrades.

What if downgrades only kick in after the current period ends? For example, if I'm on the cloud music family plan and I downgrade to the single plan, the downgrade happens at the end of the current month/cycle. Then the single plan kicks in. Basically I don't get credit for unused time. I use up the time I've paid for, then move to the lower plan.

**Comment by pippinsplugins** [comment]
@mindctrl I think you're on to something. Any other method to me just feels a bit wrong if we're not also tracking account credit.

**Comment by carter2422** [comment]
Agreed, that seems like a pretty good solution @mindctrl. The current issue is not acceptable because it can result in customers money effectively disappearing with no recourse other than manual adjustments/refunds from support.

**Comment by pippinsplugins** [comment]
To help build this, we should introduce helper functions that detect `is_upgrade()` and `is_download()` during registration.

**Comment by carter2422** [comment]
I assume you mean `is_downgrade()`? :)

**Comment by pippinsplugins** [comment]
Yep :)
